x-pack/plugins/observability/server/services/slo/transform_generators/apm_transaction_error_rate.ts

```diff
@@ -18,65 +19,58 @@ import {
   SLO_INGEST_PIPELINE_NAME,
   getSLOTransformId,
 } from '../../../assets/constants';
-import {
-  apmTransactionErrorRateSLOSchema,
-  APMTransactionErrorRateSLO,
-  SLO,
-} from '../../../types/models';
 
 const APM_SOURCE_INDEX = 'metrics-apm*';
-const ALLOWED_STATUS_CODES = ['2xx', '3xx', '4xx', '5xx'];
-const DEFAULT_GOOD_STATUS_CODES = ['2xx', '3xx', '4xx'];
 
 export class ApmTransactionErrorRateTransformGenerator implements TransformGenerator {
   public getTransformParams(slo: SLO): TransformPutTransformRequest {
-    if (!apmTransactionErrorRateSLOSchema.is(slo)) {
+    if (slo.indicator.type !== 'slo.apm.transaction_error_rate') {
       throw new Error(`Cannot handle SLO of indicator type: ${slo.indicator.type}`);
     }
 
     return getSLOTransformTemplate(
       this.buildTransformId(slo),
-      this.buildSource(slo),
+      this.buildSource(slo, slo.indicator),
       this.buildDestination(),
       this.buildGroupBy(),
-      this.buildAggregations(slo)
+      this.buildAggregations(slo.indicator)
     );
   }
 
-  private buildTransformId(slo: APMTransactionErrorRateSLO): string {
+  private buildTransformId(slo: SLO): string {
     return getSLOTransformId(slo.id);
   }
 
-  private buildSource(slo: APMTransactionErrorRateSLO) {
+  private buildSource(slo: SLO, indicator: APMTransactionErrorRateIndicator) {
```

Comment:
💬 I have to provide the slo and the indicator separately otherwise the `slo.indicator` type cannot be inferred correctly. Except if I safeguard the slo.indicator.type in every private functions as done in the main function: https://github.com/elastic/kibana/pull/142154/files#diff-21eb1f91184393609c9483ec0f5c84d8cd0ee5ffb934065f338539bb79e2bad8R27.